src/services/AssetService.ts

```diff
@@ -0,0 +1,47 @@
+import { SearchResult } from '../store';
+
+export interface GroupAsset {
+    uuid: string;
+    name: string;
+}
+
+export interface Assets {
+    search(term: string): SearchResult[];
+    add(searchResult: SearchResult): void;
+    addAll(searchResults: SearchResult[]): void;
+}
+
+export class GroupAssets {
```

Comment:
what are your thoughts on logging each change to `GroupAssets`' state while in development? e.g. 

```
prevState: {
	groups: []
},
nextState: {
	groups: [{}, {}, {}],
}
```

i don't like not being able to inspect a singleton's state, but perhaps that's not a shared concern/there's a better approach than logging each change?